datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaBasedConnectorTaskMetrics.java

```diff
@@ -40,6 +40,10 @@
   public static final String NUM_TOPICS = "numTopics";
   // keeps track of how long it takes to return from poll()
   public static final String POLL_DURATION_MS = "pollDurationMs";
+  // keeps track of how long processing takes between polls
+  public static final String TIME_SPENT_BETWEEN_POLLS_MS = "timeSpentBetweenPollsMs";
+  // keeps track of how long processing and sending all events received on poll takes
+  public static final String EVENT_PROCESSING_TIME_MS = "eventProcessingTimeMs";
```

Comment:
The name is slightly misleading. should there be a total somewhere in the name to be clear that it is not per event Processing time, but the total processing time of all the events received in one poll? One more suggestion, should we average here and add (total time/ total events), so it will give clear idea how much time we are spending to process per record. There is one more metric that might be needed how much time the producer send call took even though we know it is asynchronous. We can add that in future, if we see eventProcessingTime to be the culprint.

Reply:
done, converted it to perEventProcessingTimeMs. For producer send, we already track time since send() completes and the send callback completes in the send latency metrics (EventProducer). This processing time only includes a) time taken to translate the record and b) time blocked on send() (we can block if the producer buffer is full or if we're waiting on metadata update)